### PR TITLE
8327261: Parsing test for Double/Float succeeds w/o testing all bad cases

### PR DESCRIPTION
**Pull request by naotoj**
Fixing test cases. For bad test cases, only the first case was run, and the rest were ignored.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8327261](https://bugs.openjdk.org/browse/JDK-8327261): Parsing test for Double/Float succeeds w/o testing all bad cases (**Bug** - P4)

### Reviewers
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to [4c80fe5c](https://git.openjdk.org/jdk/pull/18113/files/4c80fe5c8f1e934e0b65d08b681c56b69427ed31)
 * [Guoxiong Li](https://openjdk.org/census#gli) (@lgxbslgx - Committer) ⚠️ Review applies to [4c80fe5c](https://git.openjdk.org/jdk/pull/18113/files/4c80fe5c8f1e934e0b65d08b681c56b69427ed31)
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer)
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18113/head:pull/18113` \
`$ git checkout pull/18113`

Update a local copy of the PR: \
`$ git checkout pull/18113` \
`$ git pull https://git.openjdk.org/jdk.git pull/18113/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18113`

View PR using the GUI difftool: \
`$ git pr show -t 18113`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18113.diff">https://git.openjdk.org/jdk/pull/18113.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18113#issuecomment-1977750580)

### CODE REVIEW

**Review by jddarcy — APPROVED** [review]
Looks fine; thanks for sending out the fix.

File: test/jdk/java/lang/Double/ParseDouble.java [hunk 1]
```diff
@@ -557,10 +557,8 @@ private static void rudimentaryTest() {
     private static void testParsing(String [] input,
                                     boolean exceptionalInput) {
         for(int i = 0; i < input.length; i++) {
```
Comment (jddarcy) [inline test/jdk/java/lang/Double/ParseDouble.java:559]: As you're updating the file, I think changing the loop to an enhanced for loop would be an improvement.

File: test/jdk/java/lang/Double/ParseDouble.java [hunk 2]
```diff
@@ -557,10 +557,8 @@ private static void rudimentaryTest() {
...
-            double d;
-
             try {
-                d = Double.parseDouble(input[i]);
+                Double.parseDouble(input[i]);
                 check(input[i]);
             }
             catch (NumberFormatException e) {
```
Comment (lgxbslgx) [inline test/jdk/java/lang/Double/ParseDouble.java:564]: The `catch` clause doesn't need to stay in a new line.

File: test/jdk/java/lang/Float/ParseFloat.java [hunk 3]
```diff
@@ -277,10 +277,8 @@ private static void rudimentaryTest() {
...
                                     boolean exceptionalInput) {
         for(int i = 0; i < input.length; i++) {
-            double d;
-
             try {
-                d = Float.parseFloat(input[i]);
+                Float.parseFloat(input[i]);
                 check(input[i]);
             }
             catch (NumberFormatException e) {
```
Comment (lgxbslgx) [inline test/jdk/java/lang/Float/ParseFloat.java:284]: The `catch` clause doesn't need to stay in a new line.

**Review by lgxbslgx — APPROVED** [review]
Looks good.

Only a trivial suggestion.

**Comment by naotoj** [comment]
Thanks for the reviews. Updated as suggested

**Review by JoeWang-Java — APPROVED** [review]

**Review by justin-curtis-lu — APPROVED** [review]

**Review by bplb — APPROVED** [review]
+1

**Review by LanceAndersen — APPROVED** [review]

**Comment by naotoj** [comment]
Thanks for the reviews!
/integrate

**Pull request closed without merging**